nf_core/modules/lint/module_tests.py

```diff
@@ -162,22 +191,43 @@ def module_tests(_, module: NFCoreComponent, allow_missing: bool = False):
                                             snap_file,
                                         )
                                     )
-                            if "versions" in str(snap_content[test_name]) or "versions" in str(snap_content.keys()):
-                                module.passed.append(
-                                    (
-                                        "test_snap_versions",
-                                        "versions found in snapshot file",
-                                        snap_file,
+                                if "versions" in str(snap_content[test_name]) or "versions" in str(snap_content.keys()):
+                                    module.passed.append(
+                                        (
+                                            "test_snap_versions",
+                                            "versions found in snapshot file",
+                                            snap_file,
+                                        )
                                     )
-                                )
-                            else:
-                                module.failed.append(
-                                    (
-                                        "test_snap_versions",
-                                        "versions not found in snapshot file",
-                                        snap_file,
+                                    # Check if version content is actual content vs MD5/SHA hash
+                                    # Related to: https://github.com/nf-core/modules/issues/6505
+                                    # Ensures version snapshots contain actual content instead of hash values
+                                    if _contains_version_hash(snap_content[test_name]):
```

Comment:
From the previous `if` statement, looks like the version can be a key of `snap_content`, so we whould take this option into account too.